src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -1142,6 +1142,28 @@ static smt_termt convert_expr_to_smt(
     mult_overflow.pretty());
 }
 
+static smt_termt convert_expr_to_smt(
+  const pointer_object_exprt &pointer_object,
+  const sub_expression_mapt &converted,
+  const smt_object_mapt &object_map)
+{
+  const auto base = pointer_object.pointer();
+  const auto type = type_try_dynamic_cast<pointer_typet>(base.type());
+  INVARIANT(
+    type, "pointer object expressions should be of pointer type");
+  const auto object = object_map.find(base);
+  INVARIANT(
+    object != object_map.end(),
+    "Objects should be tracked before converting their object ID to SMT terms");
+  const std::size_t object_id = object->second.unique_id;
+  INVARIANT(
+    object_id < power2(config.bv_encoding.object_bits),
+    "There should be sufficient bits to encode unique object identifier.");
+  const smt_termt object_bit_vector =
+    smt_bit_vector_constant_termt{object_id, config.bv_encoding.object_bits};
```

Comment:
:no_entry_sign: We cannot guarantee that the pointer a `pointer_object_exprt` is applied to will always point to the same constant object. Therefore the logic you have used here must be incorrect. I am going do some manual testing to put together and example where what you have implemented does not work as it should.